File: 0769_max_chunks_to_make_sorted.py
# https://leetcode-cn.com/problems/max-chunks-to-make-sorted/
class Solution:
    def maxChunksToSorted(self, arr):
        """
        Inputs:
            arr: list[int]
        Outputs:
            res: int
        """

        max_ = 0
        res = 0
        for i in range(len(arr)):
            if max_ < arr[i]:
                max_ = arr[i]

            if max_ == i:
                res = res + 1

        return res

File: Cpp/0769_max_chunks_to_make_sorted.cpp
// https://leetcode.cn/problems/max-chunks-to-make-sorted/

class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int cur_max = 0;
        int res = 0;
        for (int i = 0; i < arr.size(); ++i) {
            int cur = arr[i];
            if (cur > cur_max) {
                cur_max = cur;
            }
            if (cur_max == i) {
                res = res + 1;
            }
        }
        return res;
    }
};
